src/hooks/pool/usePoolTokensBalance.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+import { useRootStore } from 'src/store/root';
+import { POLLING_INTERVAL, QueryKeys } from 'src/ui-config/queries';
+import { useSharedDependencies } from 'src/ui-config/SharedDependenciesProvider';
+
+export const usePoolTokensBalance = () => {
+  const { poolTokensBalanceService } = useSharedDependencies();
+  const currentMarketData = useRootStore((store) => store.currentMarketData);
+  const user = useRootStore((store) => store.account);
+  const lendingPoolAddressProvider = currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER;
+  return useQuery({
+    queryFn: () =>
+      poolTokensBalanceService.getPoolTokensBalances({ user, lendingPoolAddressProvider }),
+    queryKey: [
+      QueryKeys.POOL_TOKENS,
+      user,
+      lendingPoolAddressProvider,
+      poolTokensBalanceService.toHash(),
```

Comment:
I think this also should have a key on lendingPool, since `poolTokensBalanceService.toHash()` is the chainID you could still have a collision if the same LPAP is used for 2 markets on a chain such as V2 ETH and V2 ETH AMM

Reply:
Other than that it looks fantastic!